cargo-dist/src/tasks.rs

```diff
@@ -137,6 +137,12 @@ pub struct BinaryIdx(pub usize);
 /// It also allows us to report what *should* happen without actually doing it.
 #[derive(Debug)]
 pub struct DistGraph {
+    /// Unique id for the system we're building on.
+    ///
+    /// Since the whole premise of cargo-dist is to invoke it once on each machine, and no
+    /// two machines have any reason to have the exact same CLI args for cargo-dist, we
```

Comment:
This makes sense but we should definitely validate it.

Reply:
The big place where this falls over right now is `all:` which ends up being every invocation of `plan/host`. in my original initial plan i was going to have it be `{cmd}:{artifacts}:{targets}` but this is *slightly* annoying since the calling command isn't really passed down. in practice it doesn't matter since plan/host makes no artifacts... hmm

Reply:
If I understand correctly, inserting something else with an identical system ID is going to lead to confusing issues down the line and we'll read inconsistent data back out of the manifest earlier. We should try to make sure the IDs are actually unique at the point where we merge manifests.